tests/ApiAuthTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\User\Tests;
+
+use HttpSoft\Message\Response;
+use HttpSoft\Message\ResponseFactory;
+use HttpSoft\Message\ServerRequest;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Http\Status;
+use Yiisoft\Test\Support\EventDispatcher\SimpleEventDispatcher;
+use Yiisoft\User\Tests\Support\MockArraySessionStorage;
+use Yiisoft\User\Tests\Support\MockIdentity;
+use Yiisoft\User\Tests\Support\MockIdentityRepository;
+use Yiisoft\User\CurrentUser;
+use Yiisoft\User\Method\ApiAuth;
+
+final class ApiAuthTest extends TestCase
+{
+    public function testSuccessfulAuthentication(): void
+    {
+        $user = $this->createCurrentUser();
+        $user->login(new MockIdentity('test-id'));
+        $result = (new ApiAuth($user, new ResponseFactory()))->authenticate(new ServerRequest());
+
+        $this->assertNotNull($result);
+        $this->assertSame('test-id', $result->getId());
+    }
+
+    public function testIdentityNotAuthenticated(): void
+    {
+        $user = $this->createCurrentUser();
+        $result = (new ApiAuth($user, new ResponseFactory()))->authenticate(new ServerRequest());
```

Comment:
```suggestion
        $result = (new ApiAuth($user))->authenticate(new ServerRequest());
```

There's only one argument accepted, not two.